src/api-binder.coffee

```diff
@@ -263,6 +266,32 @@ module.exports = class APIBinder
 				body: updatedFields
 			.timeout(conf.apiTimeout)
 
+	pinDevice: ({ app, commit }) =>
+		@config.get('deviceId')
+		.then (deviceId) =>
+			@resinApi.get
+				resource: 'release'
+				options:
+					filter:
+						belongs_to__application: app
+						commit: commit
+						status: 'success'
+					select: 'id'
+			.get(0)
+			.get('id')
```

Comment:
If no results are returned for whatever reason then this will throw an error along the lines of `cannot get id of undefined`,  and the `if !releaseId?` below implies that you expect that to be a possibility

Reply:
Ah yeah, I was getting confused with lodash's `_.get`, which handles undefined nicely. Will change.